### PR TITLE
Add svggepard app

### PR DESCRIPTION
**Pull request by szledan**
Added a simple svg parser (external code).
Added 'svggepard' app which is rendering SVG paths.

### CODE REVIEW

File: apps/svggepard/svggepard.cpp [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+static const bool isCollinear(const float x, const float y, const float* pts)
+{
+    const float threshold = 1.0f / float(0x1 << 20);
+    const bool first3pointCollinear = std::fabs(pts[0] * (pts[3] - y) + pts[2] * (y - pts[1]) + x * (pts[1] - pts[3])) < threshold;
+    const bool last3pointCollinear = std::fabs(pts[0] * (pts[3] - pts[5]) + pts[2] * (pts[5] - pts[1]) + pts[4] * (pts[1] - pts[3])) < threshold;
+    return first3pointCollinear && last3pointCollinear;
+}
+
+static void convertRawDataToRGBA(const uint32_t raw, int& red, int& green, int& blue, float& alpha)
```
Comment (kkristof) [inline apps/svggepard/svggepard.cpp]: I think we should introduce utility functions provided by the gepard for these in the future.

File: apps/svggepard/svggepard.cpp [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+            }
+
+            // Fill the path.
+            ctx.fill(); //! \todo: filltype: fill("nonzero"), fill("evenodd")
+        }
+
+        // Parse stroke style and stroke().
+        if (shp->stroke.type != NSVG_PAINT_NONE) {
+            // Parse stroke color style.
+            switch (shp->stroke.type) {
```
Comment (kkristof) [inline apps/svggepard/svggepard.cpp]: This switch almost the same as the previous, the only difference is the fill/stroke, I think this should be moved in a  function.

File: apps/svggepard/svggepard.cpp [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+            // Parse stroke line width.
+            ctx.lineWidth = shp->strokeWidth;
+
+            //! \todo: strokeDashOffset
+            //! \todo: strokeDashArray
+            //! \todo: strokeDashCount
+
+            // Parse join's type of lines.
+            switch (shp->strokeLineJoin) {
+            case NSVG_JOIN_MITER: ctx.lineJoin = "miter"; break;
```
Comment (kkristof) [inline apps/svggepard/svggepard.cpp]: This style seems wrong to me. Do we allow these one-liners? Consider the following example where we doesn't allow it and it is very similar to this:
```
if (shp->strokeLineJoin == NSVG_JOIN_MITER) ctx.lineJoin = "miter";
```

File: apps/svggepard/svggepard.cpp [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+
+            // Draw the stroke of path.
+            ctx.stroke();
+        }
+
+        shp = shp->next;
+    }
+}
+
+// Define a very simple argument parser.
```
Comment (kkristof) [inline apps/svggepard/svggepard.cpp]: It is simple, but it is a bit ugly, could you add some delimiters to make it readable?

File: apps/svggepard/svggepard.cpp [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+    NSVGimage* pImage = nsvgParseFromFile(a_svgFile.c_str(), "px", 96);
+    if (!pImage) {
+        std::cerr << "Wrong SVG file or not exist." << std::endl;
+        return 1;
+    }
+
+    // Set surface and initial the Gepard context.
+    const uint width = pImage->width;
+    const uint height = pImage->height;
+    gepard::Surface* surface = a_png.isOn ? reinterpret_cast<gepard::Surface*>(new gepard::PNGSurface(width, height))
```
Comment (kkristof) [inline apps/svggepard/svggepard.cpp]: I might be wrong, but I think there is no need for the `reinterpret_cast` here.

File: apps/svggepard/svggepard.cpp (hunk 1 above)
Comment (szledan) [reply apps/svggepard/svggepard.cpp]: I think it is a nanosvg specific converter here, so it does not have to be part of Gepard (Canvas2D).
Though you may be right.

File: apps/svggepard/svggepard.cpp (hunk 2 above)
Comment (szledan) [reply apps/svggepard/svggepard.cpp]: Even the context (Gpard) functions are different.

Comment (kkristof) [reply apps/svggepard/svggepard.cpp]: Please consider something like this:
```
enum Type {
    fill,
    stroke,
}
setColor(r, g, b, a, Type type, ctx) {
  switch(type)
  case fill:
      ctx.setFillColor(r, g, b, a)
      break
  case storke:
      ctx.setStrokeColor(r, g, b, a)
      break
  default:
      assert
}

parseElement(NSVGpaint paint, Type type, ctx) {
    if (paint.type != NSVG_PAINT_NONE) {
        switch (paint.type) {
        case NSVG_PAINT_COLOR:
             convertRawDataToRGBA(paint.color, red, green, blue, alpha);
             setColor(red * shp->opacity, green * shp->opacity, blue * shp->opacity, alpha * shp->opacity, type, ctx);
             break;
        case NSVG_PAINT_LINEAR_GRADIENT:
             /*! \todo: unipmlemnted */;
             break;
        case NSVG_PAINT_RADIAL_GRADIENT:
             /*! \todo: unipmlemnted */;
             break;
        case NSVG_PAINT_NONE:
        default:
                break;
    }     
}

parseElement(shp->fill, fill, ctx)
parseElement(shp->stroke, stroke, ctx)
```

**Comment by kkristof** [comment]
Oh, and the alpha pre-multiplication can be moved in the `setColor` function

**Pull request merged**